src/skmatter/preprocessing/_data.py

```diff
@@ -154,17 +154,16 @@ def fit(self, X, y=None, sample_weight=None):
             self.mean_ = np.zeros(self.n_features_in_)
 
         self.scale_ = 1.0
-        if self.with_std:
-            X_mean = np.average(X, weights=sample_weight, axis=0)
-            var = np.average((X - X_mean) ** 2, weights=sample_weight, axis=0)
 
+        if self.with_std:
+            var = np.average((X - self.mean_) ** 2, weights=sample_weight, axis=0)
             if self.column_wise:
-                if np.any(var < self.atol + abs(X_mean) * self.rtol):
+                if np.any(var < self.atol + abs(self.mean_) * self.rtol):
```

Comment:
I checked with scikit-learn: The way it was before is actually conform wtih sklearn's StandardScaler which makes sense also to me because the parameter is called `with_std` so standard deviation includes the mean. Can you do it just manually for your code?

Reply:
Ok, will do. This scaling choice was a heuristic one, where we found the scaling to pseudo standard deviation (compute by forcing the mean at **0**) to be the most numerically stable in the analyses that we were performing.